docs/extending_tilt_with_bazel_support.md

```diff
@@ -0,0 +1,289 @@
+---
+tilt: Extending Tilt With No Pull Requests
+layout: docs
+---
+
+# Extending Tilt with No Pull Requests
+
+There's nothing more frustrating than using a tool that works perfectly except for _one_ small thing. Even if the tool is open source it can be a daunting task to get a change accepted to it.
```

Comment:
i feel like there's so much throat-clearing here that it's really hard for a reader to answer the question "Does this doc apply to me?" That's the main question a user skimming this is trying to answer.

maybe something like

```
By default, Tilt builds images with Docker and deploys with Kubectl

But what if you don't even have a Dockerfile to give Tilt? What if you build and deploy your images with Bazel?

It's a lot simpler than you'd think. Let's see why.
```